neural_lam/utils.py

```diff
@@ -263,11 +262,65 @@ def fractional_plot_bundle(fraction):
     return bundle
 
 
-def init_wandb_metrics(wandb_logger):
+def init_wandb_metrics(wandb_logger, val_steps):
     """
     Set up wandb metrics to track
     """
     experiment = wandb_logger.experiment
     experiment.define_metric("val_mean_loss", summary="min")
-    for step in constants.VAL_STEP_LOG_ERRORS:
+    for step in val_steps:
         experiment.define_metric(f"val_loss_unroll{step}", summary="min")
+
+
+class ConfigLoader:
+    """
+    Class for loading configuration files.
+
+    This class loads a YAML configuration file and provides a way to access
+    its values as attributes.
+    """
+
+    def __init__(self, config_path, values=None):
+        self.config_path = config_path
+        if values is None:
+            self.values = self.load_config()
+        else:
+            self.values = values
+
+    def load_config(self):
+        """Load configuration file."""
+        with open(self.config_path, encoding="utf-8", mode="r") as file:
+            return yaml.safe_load(file)
+
+    def __getattr__(self, name):
+        keys = name.split(".")
+        value = self.values
+        for key in keys:
+            if key in value:
+                value = value[key]
+            else:
+                return None
+        if isinstance(value, dict):
+            return ConfigLoader(None, values=value)
+        return value
+
+    def __getitem__(self, key):
+        value = self.values[key]
+        if isinstance(value, dict):
+            return ConfigLoader(None, values=value)
+        return value
+
+    def __contains__(self, key):
+        return key in self.values
+
+    def num_data_vars(self):
+        """Return the number of data variables for a given key."""
+        return len(self.dataset.vars)
+
+    def projection(self):
```

Comment:
Maybe @property would be nice for this?

Reply:
Or maybe it would actually make more sense to pre-compute this in the constructor and store it then. So we do not have to re-instantiate it every time a plot should be created.

Reply:
Yes, I think a decorator would be a nice idea since this is an attribute of the dataset rather than something that is derived based on a set of arguments. Rather than instantiate in `__init__` I would cache with [functools.cache](https://docs.python.org/3/library/functools.html#functools.cache). Also, I think it would be nice to rename as `coords_projection` or something like that, to make it explicitly that this is about the projection of the dataset coordinates.

Reply:
great idea